metals/src/main/scala/scala/meta/internal/metals/Refactorings.scala

```diff
@@ -0,0 +1,182 @@
+package scala.meta.internal.metals
+
+import scala.concurrent.{Future, ExecutionContext}
+import scala.meta._
+import scala.meta.pc.CancelToken
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags._
+import scala.meta.internal.semanticdb.TextDocument
+import org.eclipse.{lsp4j => l}
+
+trait Refactoring {
```

Comment:
I think that the distinction only makes sense on the frontend really.  There is nothing really that distinguishes quick fixes from refactoring inside Metals.

Reply:
Although, might be good to actually have it separate and force the correct type for `refactorings`

Reply:
> force the correct type for refactorings value

what do you mean?



> I think that the distinction only makes sense on the frontend really.

I agree with this, hence my proposal to unify the interface into `CodeActionKind`

Reply:
> > force the correct type for refactorings value
> 
> what do you mean?
> 
> > I think that the distinction only makes sense on the frontend really.
> 
> I agree with this, hence my proposal to unify the interface into `CodeActionKind`

I just meant we could do:
```
def refactorings: Seq[Refactoring] =
```
to make sure everything is correctly returned and we could set the action kind for each of the refactorings. It would just kind of be a logical divide there.

Reply:
Regardless of the type, we can still "force" the code action `kind` when grouping them:

```scala
def quickfixes = List(...)
  .flatMap(_.contribute(...))
  .map(_.setKind(CodeActionKind.QuickFix))
```

I don't think we need to encode it in the type since it's a fairly innocuous mistake and the risk of getting this wrong is low.